contracts/libraries/UserLib.sol

```diff
@@ -177,66 +191,87 @@ library UserLib {
   /// @param addr Address of the user to check
   function isExists(UserCollection storage self, address addr) internal view returns (bool) {
     return self.users[addr].ipfsDoc.hash != bytes32(0x0);
   }
 
-  function updateUsersRating(UserLib.UserContext storage userContext, UserRatingChange[] memory usersRating) internal {
+  function updateUsersRating(UserLib.UserContext storage userContext, UserRatingChange[] memory usersRating, uint32 communityId) internal {
     for (uint i; i < usersRating.length; i++) {
-      updateUserRating(userContext, usersRating[i].user, usersRating[i].rating);
+      updateUserRating(userContext, usersRating[i].user, usersRating[i].rating, communityId);
     }
   }
 
   /// @notice Add rating to user
   /// @param userAddr user's rating will be change
   /// @param rating value for add to user's rating
-  function updateUserRating(UserLib.UserContext storage userContext, User storage user, address userAddr, int32 rating) internal {
+  function updateUserRating(UserLib.UserContext storage userContext, User storage user, address userAddr, int32 rating, uint32 communityId) internal {
     if (rating == 0) return;
 
-    updateRatingBase(userContext, user, userAddr, rating);
+    updateRatingBase(userContext, user, userAddr, rating, communityId);
   }
 
-  function updateUserRating(UserLib.UserContext storage userContext, address userAddr, int32 rating) internal {
+  function updateUserRating(UserLib.UserContext storage userContext, address userAddr, int32 rating, uint32 communityId) internal {
     if (rating == 0) return;
 
     User storage user = getUserByAddress(userContext.users, userAddr);
-    updateRatingBase(userContext, user, userAddr, rating);
+    updateRatingBase(userContext, user, userAddr, rating, communityId);
   }
 
-  function updateRatingBase(UserLib.UserContext storage userContext, User storage user, address userAddr, int32 rating) internal {
+  function updateRatingBase(UserLib.UserContext storage userContext, User storage user, address userAddr, int32 rating, uint32 communityId) internal {
     uint16 currentPeriod = RewardLib.getPeriod(CommonLib.getTimestamp());
-    // User storage user = getUserByAddress(userContext.users, userAddr);
-    int32 newRating = user.rating += rating;
-    uint256 pastPeriodsCount = user.rewardPeriods.length;
     
-    RewardLib.PeriodRating storage currentWeekRating = RewardLib.getUserPeriodRating(userContext.userRewards, userAddr, currentPeriod);
-    bool isFirstTransactionOnThisWeek = pastPeriodsCount == 0 || user.rewardPeriods[pastPeriodsCount - 1] != currentPeriod; 
+    CommunityRatingForUser storage communityUser = userContext.userRatingCollection.communityRatingForUser[userAddr];
+    UserRating storage userRating = communityUser.userRating[communityId];
+    if (!userRating.isActive) {
+      userRating.rating = START_USER_RATING;
+      userRating.payOutRating = START_USER_RATING;
+      userRating.isActive = true;
```

Comment:
Call `communityUser.userRewards[currentPeriod].activeInCommunity.push(communityId);` here.
No need to do any checks in the loop.